### PR TITLE
QuerySpec -> QueryParams converter

### PR DESCRIPTION
**Pull request by Ramblurr**
see #347

### CODE REVIEW

File: katharsis-core/src/main/java/io/katharsis/legacy/queryParams/DefaultQueryParamsConverter.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    private String getResourceType(Class<?> resourceClass) {
+        RegistryEntry registryEntry = resourceRegistry.getEntryForClass(resourceClass);
+        if (registryEntry == null) {
+            throw new IllegalArgumentException("resourceType for class " + resourceClass.getName() + " not found");
+        }
+        ResourceInformation resourceInformation = registryEntry.getResourceInformation();
+        return resourceInformation.getResourceType();
+    }
+
+    protected void applyFiltering(QuerySpec spec, QueryParams queryParams) {
```
Comment (chb0github) [inline katharsis-core/src/main/java/io/katharsis/legacy/queryParams/DefaultQueryParamsConverter.java]: now that we are able to use java 8 I would really prefer to see this as a couple of lambdas.

File: katharsis-core/src/main/java/io/katharsis/legacy/queryParams/QueryParams.java [hunk 2]
```diff
@@ -61,6 +62,32 @@ void setIncludedRelations(TypedParams<IncludedRelationsParams> includedRelations
     }
 
     @Override
+    public int hashCode() {
```
Comment (chb0github) [inline katharsis-core/src/main/java/io/katharsis/legacy/queryParams/QueryParams.java]: Where did you come up with this implementation of equals/hashcode ? It doesn't look like the intellij standard.

File: katharsis-core/src/main/java/io/katharsis/legacy/queryParams/include/Inclusion.java [hunk 3]
```diff
@@ -29,6 +29,15 @@ public String getPath() {
     }
 
     @Override
+    public int compareTo(Inclusion o) {
```
Comment (chb0github) [inline katharsis-core/src/main/java/io/katharsis/legacy/queryParams/include/Inclusion.java]: Java 8 has some really nice ways to build comparators functionally and even putting nulls first/last. At a minimum, could we have a comment on how you _think_ this sort should work?

File: katharsis-core/src/main/java/io/katharsis/legacy/queryParams/params/FilterParams.java [hunk 4]
```diff
@@ -16,6 +18,21 @@ public FilterParams(Map<String, Set<String>> params) {
     }
 
     @Override
+    public int hashCode() {
```
Comment (chb0github) [inline katharsis-core/src/main/java/io/katharsis/legacy/queryParams/params/FilterParams.java]: Please use the standard hash/equals generators from IJ - I would preferr wholesale move to Lombok

File: katharsis-core/src/main/java/io/katharsis/legacy/queryParams/QueryParams.java (hunk 2 above)
Comment (Ramblurr) [inline katharsis-core/src/main/java/io/katharsis/legacy/queryParams/QueryParams.java]: IntelliJ did generate it. I have 2016.3.5

File: katharsis-core/src/main/java/io/katharsis/legacy/queryParams/DefaultQueryParamsConverter.java [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+                String path = joinPath(includedField.getAttributePath());
+                pathSet.add(path);
+            }
+            IncludedFieldsParams includedFieldsParams = new IncludedFieldsParams(pathSet);
+            decodedSparseMap.put(resourceType, includedFieldsParams);
+        }
+        queryParams.setIncludedFields(new TypedParams<>(Collections.unmodifiableMap(decodedSparseMap)));
+    }
+
+    protected void applyRelatedFields(QuerySpec spec, QueryParams queryParams) {
```
Comment (chb0github) [inline katharsis-core/src/main/java/io/katharsis/legacy/queryParams/DefaultQueryParamsConverter.java]: All of this is much too verbose and needs to be reduced.

**Comment by Ramblurr** [comment]
@chb0github This implementation was largely based on @remmeier (`DefaultQuerySpecConverter`](https://github.com/katharsis-project/katharsis-framework/blob/2d0f58dc20c368291333e98f29804e4704a9f163/katharsis-core/src/main/java/io/katharsis/legacy/internal/DefaultQuerySpecConverter.java).

Feel free to suggest how you think the code would look with lambadas, I would appreciate that.

I'll refactor the comparator and equals implementation, though not the hashcode as it is already from intellij.

File: katharsis-core/src/main/java/io/katharsis/legacy/queryParams/params/FilterParams.java (hunk 4 above)
Comment (Ramblurr) [inline katharsis-core/src/main/java/io/katharsis/legacy/queryParams/params/FilterParams.java]: This is standard hashCode. Equals have all been updated.

**Comment by Ramblurr** [comment]
@chb0github Also, I started using Java 8 features in this branch and maven complains that the language level is still set to 1.7. 

I rebased locally on master to catch any updates, but katharsis-core is still using 1.7.

Anyways, I just pushed new version that implements equals+hashcode with the default IntelliJ impl.

**Comment by chb0github** [comment]
I am [looking at master](https://github.com/katharsis-project/katharsis-framework/blob/master/pom.xml#L20) right now it says 1.8.

**Pull request merged**